core/src/main/scala/org/apache/spark/ui/jobs/JobsTab.scala

```diff
@@ -35,4 +37,18 @@ private[ui] class JobsTab(parent: SparkUI) extends SparkUITab(parent, "jobs") {
 
   attachPage(new AllJobsPage(this))
   attachPage(new JobPage(this))
+
+  def handleKillRequest(request: HttpServletRequest): Unit = {
+    if (killEnabled && (parent.securityManager.checkModifyPermissions(request.getRemoteUser))) {
+      val killFlag = Option(request.getParameter("terminate")).getOrElse("false").toBoolean
+      val jobId = Option(request.getParameter("id")).getOrElse("-1").toInt
+      if (jobId >= 0 && killFlag && jobProgresslistener.activeJobs.contains(jobId)) {
+        sc.get.cancelJob(jobId)
+      }
```

Comment:
And if the Job isn't actually going to be canceled, then there is no need to delay the page refresh (which I'm not entirely happy with, but I'm not going to try to resolve that issue right now.)  So...
```scala
sc.foreach { sparkContext =>
  sparkContext.cancelJob(id)
  Thread.sleep(100)
}
```

And we better make that `jobId` is an `Option[Int]` instead of an `Option[String]`, so...
```scala
val jobId = Option(request.getParameter("id")).map(_.toInt)
```